org.adempiere.base/src/org/compiere/model/MMenu.java

```diff
@@ -42,6 +45,26 @@
 	 */
 	private static final long serialVersionUID = -6671861281736697100L;
 
+	/** Cache for MMenu */
+	private static CCache<Integer, MMenu>	cache_menu			= new CCache<Integer, MMenu>(MMenu.Table_Name, "CacheMenu", 50);

```

Comment:
Question for @hengsin:

Several classes in this commit are implementing CCache.

Are there new guidelines to implement CCache in core?  (example implement interface ImmutablePOSupport?)

Reply:
Do I needs to refer https://groups.google.com/g/idempiere/c/9CRxVUwt18s/m/tE9kTJpUBAAJ?

Reply:
@dpansheriya , yes, please follow the pattern in current source (there are plenty of them).

Reply:
only MMenu cache is caching PO, So added immutable PO support on MMenu and converted cache to immutable.